src/string_scanner.cr

```diff
@@ -280,6 +282,34 @@ class StringScanner
     @str.byte_slice(@byte_offset, @str.bytesize - @byte_offset)
   end
 
+  # Returns one byte from the current offset.
+  # ```
+  # require "string_scanner"
+  #
+  # s = StringScanner.new("あ")
+  # s.read_byte # => "\xE3"
+  # s.read_byte # => "\x81"
+  # s.read_byte # => "\x82"
+  # ```
+  def read_byte : String?
```

Comment:
In what situation is this something that is reasonable (rather than reading a full utf8 character)? The only situations I can think of is where the data isn't actually a valid string, and I'd argue that if that is the case a solution working directly on `Slice` would be more appropriate.

Reply:
I also think that if we go with this, we should return `UInt8?`, not `String?`

But yes, it would be nice to know the actual use case for this.

Reply:
changed to return `UInt8?`.